packages/cli/src/site-configs.types.ts

```diff
@@ -9,9 +9,9 @@ export type BaseSiteConfig = {
     preloginEnabled?: boolean;
     public?: Record<string, unknown>;
 };
-export type ExtractPrivateSiteConfig<S extends BaseSiteConfig> = S & {
+export type ExtractPrivateSiteConfig<S extends BaseSiteConfig> = Omit<S, "public"> & {
     url: string;
-};
-export type ExtractPublicSiteConfig<S extends BaseSiteConfig> = Pick<S, "name" | "domains" | "preloginEnabled" | "public"> & {
+} & S["public"];
+export type ExtractPublicSiteConfig<S extends BaseSiteConfig> = Pick<S, "name" | "domains" | "preloginEnabled"> & {
     url: string;
-};
+} & S["public"];
```

Comment:
I suppose `S["public"]` should be first because it's also first in https://github.com/vivid-planet/comet/pull/2297/files and overridden by the other props